communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzer.java

```diff
@@ -0,0 +1,219 @@
+
+                        // convert from ProjectVersionRef to GAV
+                        GAV originGAV = generateGAV(origin);
+                        GAV targetGAV = generateGAV(target);
+
+                        // get the origin ScmGAVRelationship
+                        ScmGAVRelationship originScmGAVRel = addGAVModuleToMapper(gavSCMRelMap,
+                                originGAV);
+
+                        addTargetToScmGAVRelationship(originScmGAVRel, targetGAV, relationship);
+                    }
+                }
+
+                return new LinkedList<>(gavSCMRelMap.values());
+
+            } finally {
+                // cleanup
+                FileUtils.deleteDirectory(tempDir);
+            }
+        } catch (TransferException | GalleyMavenException | URISyntaxException | CartoDataException
+                | IOException e) {
+            throw new PomAnalysisException(e);
+        }
+    }
+
+    /**
+     * Return an existing ScmGAVRelationship if present in the map, otherwise create a new ScmGAVRelationship
+     * and map the GAV to the ScmGAVRelationship
+     *
+     * @param map
+     * @param gav
+     * @return
+     */
+    private ScmGAVRelationship addGAVModuleToMapper(Map<GAV, ScmGAVRelationship> map, GAV gav) {
+        return map.computeIfAbsent(gav, (k) -> new ScmGAVRelationship(k));
+    }
+
+    private GAV generateGAV(ProjectVersionRef ref) {
+        return new GAV(ref.getGroupId(), ref.getArtifactId(), ref.getVersionString());
+    }
+
+    private DiscoveryResult getDiscoveryResult(File tempDir, CartographerCore carto,
+            ProjectVersionRef ref) throws GalleyMavenException, URISyntaxException,
+            CartoDataException {
+
+        MavenPomReader pomReader = carto.getGalley().getPomReader();
+        MavenPomView pomView = pomReader.read(ref, getRepoLocations(tempDir));
+
+        Location location = new SimpleLocation("file:" + tempDir.getAbsolutePath());
+        URI src = new URI(location.getUri());
+
+        DiscoveryConfig disConf = new DiscoveryConfig(src);
+        disConf.setIncludeBuildSection(false);
+        disConf.setIncludeManagedDependencies(false);
+        disConf.setIncludeManagedPlugins(false);
+
+        MavenModelProcessor processor = new MavenModelProcessor();
+        return processor.readRelationships(pomView, src, disConf);
+    }
+
+    // TODO: get the list of locations from main pom.xml
+    private List<? extends Location> getRepoLocations(File cacheDir) {
+        Location location = new SimpleLocation("file:" + cacheDir.getAbsolutePath());
+        List<? extends Location> repoLocations = Arrays.asList(location, new SimpleLocation(
+                "central", "http://repo.maven.apache.org/maven2/"), new SimpleLocation("eap",
+                "http://maven.repository.redhat.com/techpreview/all"), new SimpleLocation(
+                "sonatype", "https://oss.sonatype.org/content/repositories"));
```

Comment:
This is used to fetch the parent pom information that is sometimes defined in the root pom.xml. Right now these values are hardcoded, but I'd like to parse the pom.xmls in the future to extract the information about the repositories used.